service/src/test/java/com/derpgroup/quip/manager/QuipManagerTest.java

```diff
@@ -72,4 +79,65 @@ public void testConversationRequest_another_noConversationHistory_noBot(){
     assertTrue(outputRawText.length() > 0);
     assertEquals(outputRawText,"I don't know how to handle requests for unnamed bots.");
   }
+  
+  @Test
+  public void testDetermineMaxQuipHistorySize(){
+    assertEquals(manager.determineMaxQuipHistorySize(16),8);
+    assertEquals(manager.determineMaxQuipHistorySize(10),5);
+    assertEquals(manager.determineMaxQuipHistorySize(4),2);
+  }
+  
+  @Test
+  public void testConversationRequest_noRepeats(){
+    
+    QuipMetadata quipMetadata = new QuipMetadata();
+    quipMetadata.setConversationHistory(new ArrayDeque<ConversationHistoryEntry>());
+    Deque<String> originalComplimentUsed = new ArrayDeque<String>();
+    for(int i=0; i<Compliments.values().length/2; i++){
+      originalComplimentUsed.add(Compliments.values()[i].name());
+    }
+    Deque<String> originalWinsultsUsed = new ArrayDeque<String>();
+    for(int i=0; i<Winsults.values().length/2; i++){
+      originalWinsultsUsed.add(Winsults.values()[i].name());
+    }
+    Deque<String> originalInsultsUsed = new ArrayDeque<String>();
+    for(int i=0; i<Insults.values().length/2; i++){
+      originalInsultsUsed.add(Insults.values()[i].name());
+    }
+    Deque<String> originalBackhandedComplimentsUsed = new ArrayDeque<String>();
+    for(int i=0; i<BackhandedCompliments.values().length/2; i++){
+      originalBackhandedComplimentsUsed.add(BackhandedCompliments.values()[i].name());
+    }
+    
+    for(int i=0; i<20; i++){
```

Comment:
This is overkill.  The odds of us failing to trigger a repeat if we had a bug is 5.96 x 10^-8.

Lower the run count (and remember that it's currently 4x the number of iterations)